shadowsocks/replay.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shadowsocks
+
+import (
+	"encoding/binary"
+	"sync"
+)
+
+// Capacities in excess of 20,000 are not recommended, due to the false
+// positive rate of up to 2 * capacity / 2^32 = 1 / 100,000.  If larger
+// capacities are desired, the key type should be changed to uint64.
+const maxCapacity = 20_000
+
+type empty struct{}
+
+// IVCache allows us to check whether an initialization vector was among
+// the last `capacity` IVs.  It requires approximately 15*capacity bytes
+// of memory (4 bytes per key plus 10.79 bytes of overhead:
+// https://go.googlesource.com/go/+/refs/tags/go1.13.5/src/runtime/map.go#43).
+// The zero value is a cache with capacity 0, i.e. no cache.
+type IVCache struct {
```

Comment:
We have a terminology issue here. The Shadowsocks documentation refers to the IV as salt, and so does our existing code. Let's stick to a consistent terminology and change it to Salt for readability

Reply:
Done